File: provider/src/main/java/my/sample/provider/impl/HashDecorator.java
package my.sample.provider.impl;

import my.sample.provider.SampleDecorator;

/**
 * @author mohammad shamsi <m.h.shams@gmail.com>
 */
public class HashDecorator implements SampleDecorator {
    @Override
    public String decorate(String message) {
        return "### " + message + " ###";
    }
}

File: provider/src/main/kotlin/my/sample/provider/impl/HashDecorator.kt
package my.sample.provider.impl

import my.sample.provider.SampleDecorator

/**
 * @author mohammad shamsi <m.h.shams@gmail.com>
 */
public class HashDecorator : SampleDecorator {

    override fun decorate(message: String) = "### $message ###"
}
